lib/config/presets/gitlab/index.ts

```diff
@@ -52,7 +53,73 @@ export async function fetchJSONFile(
     throw new Error(PRESET_DEP_NOT_FOUND);
   }
 
-  return parsePreset(res.body, fileName);
+  // Extract the actual filename without path or query parameters
+  const cleanFileName = extractFilenameFromGitLabPath(fileName);
+
+  return parsePreset(res.body, cleanFileName);
+}
+
+/**
+ * Extracts the actual filename from a GitLab path, which could be:
+ * - A simple filename: "renovate.json5"
+ * - A path: "path/to/renovate.json5"
+ * - A URL-encoded path: "src%2Frenovate%2Fconfig.json5"
```

Comment:
when this will happen?

Reply:
there is no test case for the full url which by the way doesn't use that code path